geode-for-redis/src/main/java/org/apache/geode/redis/internal/data/RedisSet.java

```diff
@@ -108,21 +104,18 @@ public static int sdiffstore(RegionProvider regionProvider, RedisKey destination
   private static MemberSet calculateDiff(RegionProvider regionProvider, List<RedisKey> keys,
       boolean updateStats) {
     RedisSet firstSet = regionProvider.getTypedRedisData(REDIS_SET, keys.get(0), updateStats);
+    MemberSet diff = new MemberSet();
     if (firstSet.scard() == 0) {
```

Comment:
Since we never store a empty RedisSet (if it is empty it just gets removed) scard will never return 0. You should delete this if block.

Reply:
I think that if there is no value associated with the key then we return a `NULL_REDIS_SET` here, which returns 0 when `scard()` is called on it, so this check is actually doing something.

Reply:
I changed it to NULL_REDIS_SET, because I thought it would make it easier to understand. Let me know if I should change it back.